framework/src/main/java/org/tron/core/db/Manager.java

```diff
@@ -756,12 +759,16 @@ private void switchFork(BlockCapsule newHead)
       TooBigTransactionException, TooBigTransactionResultException, DupTransactionException,
       TransactionExpirationException, NonCommonBlockException, ReceiptCheckErrException,
       VMIllegalException, ZksnarkException, BadBlockException {
+
+    MetricsUtil.meterMark(MetricsKey.BLOCKCHAIN__FORK_COUNT, 1);
+
     Pair<LinkedList<KhaosBlock>, LinkedList<KhaosBlock>> binaryTree;
     try {
       binaryTree =
           khaosDb.getBranch(
               newHead.getBlockId(), getDynamicPropertiesStore().getLatestBlockHeaderHash());
     } catch (NonCommonBlockException e) {
+      MetricsUtil.meterMark(MetricsKey.BLOCKCHAIN_FAIL_FORK_COUNT, 1);
```

Comment:
It's also counted in finally. It's counted repeatedly

Reply:
There are total two try catch. Once a exception shows at first try catch, then will throw e , the later code will not execute, which will not execute second finally function since this exception will not be captured and also throw e.